techtonica-projects/contact-app/client/src/App.test.jsx

```diff
@@ -0,0 +1,29 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import { expect, test, vi } from 'vitest';
+import App from './App';
+
+
+
+
+
+  test('renders contact list fetched from database', async () => {
+      // ARRANGE:
+  global.fetch = vi.fn(() =>
+    Promise.resolve({
+      json: () =>
+        Promise.resolve([
+            { name: 'Harry Potter', email: 'test@hogwarts.com', phone: '123456789' }
+        ]),
+    })
+  );
+
+  render(<App />);
+
+  // ACT & ASSERT:
+  await waitFor(() => {
+    expect(global.fetch).toHaveBeenCalledTimes(1);
```

Comment:
is there scope to further ascertain that client side variables are being set with the `name`, `email` and `phone` values given on line 15?

Reply:
I tried to add the code below, but was I getting an error relating to the testing configuration. I will try to figure out the issue when I get a chance. 
   ```
     expect(screen.getByText('Harry Potter')).toBeInTheDocument();
        expect(screen.getByText('test@hogwarts.com')).toBeInTheDocument();
        expect(screen.getByText('123456789')).toBeInTheDocument();
```